test/generic_container_test.exs

```diff
@@ -1,9 +1,19 @@
 defmodule Testcontainers.GenericContainerTest do
   use ExUnit.Case, async: true
 
+  import TestHelper, only: [port_open?: 1]
+
   test "can start and stop generic container" do
     config = %Testcontainers.Container{image: "redis:latest"}
     assert {:ok, container} = Testcontainers.start_container(config)
     assert :ok = Testcontainers.stop_container(container.container_id)
   end
+
+  test "can start and stop generic container with network mode set to host" do
+    config = %Testcontainers.Container{image: "redis:latest", network_mode: "host"}
+    assert {:ok, container} = Testcontainers.start_container(config)
+    Process.sleep(5000)
+    assert :ok = port_open?(6379)
```

Comment:
how will host network mode work together will exposed ports?

Reply:
When you run network mode host and also define ports the docker engine print the warning message like:

```
Published ports are discarded when using host network mode
```

Because of this, I already consider the behavior defined by the container engine itself and nothing would need to be done in the code to deal with this situation

Reply:
We can do this, especially if the tests passes ;) they can be flaky so I will check if they pass properly. 

Not sure exactly which release it will be included in, but that's for another day another time ;)

Reply:
btw if you have multiple commits they will be squashed. I try not to merge multiple commits.

Reply:
Hi thanks for responding. Sorry but I don't understand if I have to do anything?